test/test_testing.py

```diff
@@ -479,14 +478,12 @@ def test_trivial_passing_test_case_on_cpu_cuda(self, device):
 
 if __name__ == '__main__':
     run_tests()
-"""
-
-        # Test running of cuda assert test suite should early terminate.
-        p = subprocess.run([sys.executable, '-c', problematic_test_script], stderr=subprocess.PIPE, timeout=120)
+""")
         # should capture CUDA error
-        self.assertIn('CUDA error: device-side assert triggered', p.stderr.decode('ascii'))
+        self.assertIn('CUDA error: device-side assert triggered', results[1].decode('ascii'))
```

Comment:
`results[1]` is now less clear than `p.stderr`, could we change it so that instead of getting the second element of results, we use something named as well?